helpers/helpers.v2.1.d/go

```diff
@@ -171,39 +110,22 @@ ynh_go_remove() {
 #
 # usage: _ynh_go_cleanup
 _ynh_go_cleanup() {
-    # List required Go versions
-    local installed_apps=$(yunohost app list --output-as json --quiet | jq -r .apps[].id)
-    local required_go_versions=()
-    for installed_app in $installed_apps; do
-        local installed_app_go_version=$(ynh_app_setting_get --app="$installed_app" --key="go_version")
-        if [[ $installed_app_go_version ]]; then
-            required_go_versions+=("$installed_app_go_version")
-        fi
-    done
 
     # Remove no more needed Go versions
-    local installed_go_versions=$(goenv versions --bare --skip-aliases | grep -Ev '/')
+    local installed_go_versions=$($GOENV_ROOT/bin/goenv versions --bare --skip-aliases | grep -Ev '/')
     for installed_go_version in $installed_go_versions; do
-        if ! (printf '%s\0' "${required_go_versions[@]}" | grep -Fxqz -- "$installed_go_version"); then
-            ynh_print_info "Removing of Go-$installed_go_version"
-            $GOENV_INSTALL_DIR/bin/goenv uninstall --force "$installed_go_version"
+        if ! grep -qE "^go_version: '?$installed_go_version'?" /etc/yunohost/apps/*/settings.yml
+        then
+            ynh_print_info "Removing Go-$installed_go_version"
+            $GOENV_ROOT/bin/goenv uninstall --force "$installed_go_version"
         fi
     done
 
-    # If none Go version is required
-    if ((${#required_go_versions[@]} == 0)); then
+    # If no app uses Go anymore
+    if ! grep -q "^go_version:" /etc/yunohost/apps/*/settings.yml; then
         # Remove goenv environment configuration
-        ynh_print_info "Removing of goenv"
-        ynh_safe_rm "$GOENV_INSTALL_DIR"
+        ynh_print_info "Removing goenv"
```

Comment:
```suggestion
```
I find GO helper unnecessarily talkative

Reply:
Ugh yeah indeed especially since at the end of the day I think it's only downloading a 50~100 MB file and not compiling Go locally ?